tangchengjin/demo.py

```diff
@@ -2,37 +2,63 @@
 from random import choice
 name = input("Input file name: ")  ## optional: loop
 ## 1.1. try.. except check file exists
+flag = 0
```

Comment:
It's of better style to use a variable of boolean value with a more informative name.
E.g., `validInput = False`
Or, alternatively, you can leave out the flag, use an infinite loop, and `break` out of the loop once input is valid.